pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -271,6 +271,25 @@ func (a Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSpe
 
 	kclient.AddBootstrapSupervisordInitContainer(podTemplateSpec)
 
+	// if there are preStart events, add them as init containers to the podTemplateSpec
+	preStartEvents := a.Devfile.Data.GetEvents().PreStart
+	if len(preStartEvents) > 0 {
+		var eventCommands []string
+		commandsMap := common.GetCommandsMap(a.Devfile.Data.GetCommands())
+		containersMap := utils.GetContainersMap(containers)
+
+		for _, event := range preStartEvents {
+			eventSubCommands := common.GetCommandsFromEvent(commandsMap, strings.ToLower(event))
+			eventCommands = append(eventCommands, eventSubCommands...)
+		}
+
+		klog.V(4).Infof("preStart events commands are: %v", strings.Join(eventCommands, ","))
+		utils.AddPreStartEventInitContainer(podTemplateSpec, commandsMap, eventCommands, containersMap)
+		if len(eventCommands) > 0 {
+			log.Successf("preStart commands will be executed during pod startup: %s", strings.Join(eventCommands, ","))
```

Comment:
Let's use a simpler term like `Component startup`

Reply:
Also can we start with a capital `PreStart`?

Reply:
This seems to be printed after every `push`. I think we should print this only once during component start up.

Reply:
- Not sure about `Component startup` see third point below.

- Can update to `PreStart`

- Nope it is correct, whenever a pod starts up, the prestart events will be executed because they're initContainers of the pod. Kubernetes executes initContainers everytime a pod starts/restarts.

Reply:
But it seems the pod is restarted upon every push if the devfile has `PreStart` events. I feel something in the deployment is getting changed on every push.